pkg/cmd/experimental/buildchain/buildchain.go

```diff
@@ -94,12 +94,20 @@ func (o *BuildChainOptions) Complete(f *clientcmd.Factory, cmd *cobra.Command, a
 	}
 	o.c, o.t = oc, oc
 
-	// Parse user input
-	o.name, o.tag, err = buildChainInput(args[0])
+	kind, defaultKind := "", "ImageStreamTag"
+	mapper, _ := f.Object()
+	kind, o.name, err = osutil.ResolveResource(defaultKind, args[0], mapper)
 	if err != nil {
-		return cmdutil.UsageError(cmd, err.Error())
+		return err
+	}
+	if kind != defaultKind {
```

Comment:
change to switch to make it obvious what resources we support, and make it straightforward to add support for new resources
```
switch resource {
case "imagestreamtags":
  ... handle
default:
  ... error
}
```